ui/src/app/settings/reducers/users.js

```diff
@@ -11,14 +11,49 @@ const users = produce(
         draft.loaded = true;
         draft.items = action.payload;
         break;
+      case "UPDATE_USERS_START":
+      case "CREATE_USERS_START":
+        draft.saved = false;
+        draft.saving = true;
+        break;
+      case "UPDATE_USERS_ERROR":
+      case "CREATE_USERS_ERROR":
+        draft.errors = action.error;
+        draft.saving = false;
+        break;
+      case "UPDATE_USERS_SUCCESS":
+      case "CREATE_USERS_SUCCESS":
+        draft.errors = {};
+        draft.saved = true;
+        draft.saving = false;
+        break;
+      case "UPDATE_USER_SYNC":
+        for (let i in draft.items) {
+          if (draft.items[i].id === action.payload.id) {
+            draft.items[i] = action.payload;
+            break;
```

Comment:
I've updated the config reducer to match this one. This one is more performant as it breaks once it finds the item to update instead of mapping over all items.